### DISCUSSION TITLE
Problem with 2 different Controllers

### DISCUSSION THREAD

**Discussion by roemu** [devcompl/TTController#122] [Configuration] [answered]
Hello there
I just bought a second set of 5x TT RGB Riing PLUS 12 Fans and I got a second controller in the box. My first controller is a Riing Plus Controller but the second one is Riing Quad.

I have all sorts of problems with my setup now, for example: the lights of the fans aren't synced, can't change the color of the fans, Your software has a timeout error when starting, TT RGB Plus software doesnt work at all (Doesn't detect fans), when I run Debug -> Report/Controllers it shows more fans than I actually have installed and even then not the right rpm for some of them.

I have 4 fans in both controllers and they are connected via the daisy chain cable that you get from the box.
My question is if there's a problem with connecting two different controllers (Riing Plus + Riing Quad)?

**Comment by roemu** [comment]
**Update**
Here's the config.json that i'm using:

```json
{
  "Profiles": [
    {
      "Name": "Default",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4],
        [9802, 8102, 5],
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3],
        [9802, 8800, 4],
        [9802, 8800, 5]
      ],
      "Effects": {
        "Type": "FullColorEffect",
        "Config": {
          "Color": [255,255,255],
            "Trigger": {
              "Type": "OneTimeTrigger"
            }
        }
      },
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController",
          "Config": {
            "CurvePoints": [
              [30, 30],
              [45, 50],
              [55, 60],
              [65, 75],
              [75, 100]
            ],
            "Sensors": ["/intelcpu/0/temperature/8"],
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4],
        [9802, 8102, 5],
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3],
        [9802, 8800, 4],
        [9802, 8800, 5]
      ],
      "Speed": 35,
      "Colors": {
        "Full": [255, 0, 0]
      }
    }
  ],
  "PortConfigs": [],
  "SensorConfigs": [
    {
      "Sensors": ["/intelcpu/0/temperature/8"],
      "Config": {
        "CriticalValue": 90
      }
    }
  ]
}
```

When I use the option "Run in Console" it spits out an error.
[Error.txt](https://github.com/MoshiMoshi0/TTController/files/5824761/Error.txt)

**Comment by MoshiMoshi0** [comment]
The error is pretty self explanatory, `"Effects"` is an array and is missing the [] brackets, it should look similar to `"SpeedControllers"`.

> Your software has a timeout error when starting,

It probably times out because of the error, you can use console mode to see why it might timeout.

> when I run Debug -> Report/Controllers it shows more fans than I actually have installed

It shows all possible ports, there is no way to tell how many are fans installed.

> even then not the right rpm for some of them.

Not sure what is not right rpm, but make sure you are not running thermaltake software at the same time as TTController.

Also, please post the debug report.

**Comment by roemu** [comment]
Ah yes. What I meant with not the right rpm is that some of the fans show up as 0 rpm, but that is explained by that the report shows every port. I uninstalled the Thermaltake software already and I now fixed the code up a bit (found some other parts that were wrong). With the changes that I made, the code now runs. Thank you for that!
But now I have a new problem. Since last night, my second controller stopped showing up at all, as you can see in the Debug Report.
[DebugReport.txt](https://github.com/MoshiMoshi0/TTController/files/5825853/DebugReport.txt)
Could that be due to having 2 different kind of controllers (Riing PLUS, Riing Quad)?
Also I just noticed that the code just turns off all of the LED's of the first 4 fans, it doesn't do it to the other fans because it isn't detecting them. Here's my current code and the output from "Run In Console":
[Code.txt](https://github.com/MoshiMoshi0/TTController/files/5825860/Code.txt)
[RunInConsole.txt](https://github.com/MoshiMoshi0/TTController/files/5825862/RunInConsole.txt)

**Comment by MoshiMoshi0** [comment, marked as answer]
Your controller is probably faulty, see [this](https://github.com/MoshiMoshi0/TTController/discussions/106#discussioncomment-252847) similar thread.

> Also I just noticed that the code just turns off all of the LED's of the first 4 fans

Your `"StaticColorEffect"` configuration is wrong, check [this ](https://moshimoshi0.github.io/TTController/plugins/effects/staticcoloreffect) and [this](https://moshimoshi0.github.io/TTController/common/led-color-provider).

**Reply by roemu** [reply]
Ok I'm gonna give the store where I bought the fans a call and get back to you if anything changes.
